SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -272,12 +276,17 @@ void USpatialActorChannel::DeleteEntityIfAuthoritative()
 				Sender->RetireEntity(EntityId, Actor->IsNetStartupActor());
 			}
 		}
-		else
+		else if (bCreatedEntity) // We have not gained authority yet
 		{
-			// This is unsupported, and shouldn't happen, don't attempt to cleanup entity to better indicate something has gone wrong
-			UE_LOG(LogSpatialActorChannel, Error, TEXT("DeleteEntityIfAuthoritative called on actor channel with null actor - entity id (%lld)"), EntityId);
+			Actor->SetReplicates(false);
+			Receiver->RetireWhenAuthoritive(EntityId, NetDriver->ClassInfoManager->GetComponentIdForClass(*Actor->GetClass()), Actor->IsNetStartupActor(), Actor->GetTearOff()); // Ensure we don't recreate the actor
```

Comment:
```suggestion
			Receiver->RetireWhenAuthoritive(EntityId, NetDriver->ClassInfoManager->GetComponentIdForClass(*Actor->GetClass()), Actor->IsNetStartupActor(), Actor->GetTearOff()); // Cache actor state as it may be destroyed
```

Reply:
I'd argue it's not actor state you are caching. You just need to know what the entity ID is to know to ignore ops relating to that actor in future.

I.e. you want to avoid seeing the entity due to interest before you get authority, thinking it's a new actor from another server and spawning it.

Reply:
But you need to know the startup and tearoff states, I'd consider that Actor state.
Entity and class are also state related to the actor.
Seems a bit nitty, but I just don't think the existing comment is valid.